File: BASICS/perfectno.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    int sum =0;
    cout<<"eneter a number";
    cin>>n;
     for(int i=1;i<=n;i++){
        if(n%i==0){
         sum = sum + i;
        }
        }
        if (sum == n){
            cout<<"the perfect";
        }
        else {
            cout<<"not perfect";
        }
     }

File: JAVA 1D ARRAY/BASIC 40/perfectno.java
import java.util.*;

public class perfectno {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter a number: ");
        int n = sc.nextInt();
        int sum = 0;
        int original = n;
        for(int i=1;i<n;i++){
            if(n % i == 0){
                sum = sum+i;  
            }
        }
        if(original == sum)
        System.out.println("its perfect no");
        else
        System.out.println("not perfect no");
       
    }
    
}
